### ISSUE TITLE
[question] profile not found from $CONAN_USER_HOME/profiles ?

### ISSUE THREAD

**Issue by unzap** [conan-io/conan#6407]
Using Conan 1.21.1

I have custom Conan home directory:
/home/foo/conan_home_test
/home/foo/conan_home_test/.data
/home/foo/conan_home_test/profiles/test_profile
/home/foo/conan_home_test/conan.conf
/home/foo/conan_home_test/...

```
export CONAN_USER_HOME=/home/foo/conan_home_test
conan install . -pr test_profile
Auto detecting your dev setup to initialize the default profile (/home/foo/conan_home_test/.conan/profiles/default
```
It did not find the "profiles/" under the CONAN_USER_HOME and it initialized nested .conan directory?

Does it expect ".conan" to be in the CONAN_USER_HOME path?
If I change the directory stucture like:

/home/foo/conan_home_test/.conan
/home/foo/conan_home_test/.conan/.data
/home/foo/conan_home_test/.conan/profiles/test_profile
/home/foo/conan_home_test/.conan/conan.conf
/home/foo/conan_home_test/.conan/...

```
export CONAN_USER_HOME=/home/foo/conan_home_test/.conan
conan install . -pr test_profile
Auto detecting your dev setup to initialize the default profile (/home/foo/conan_home_test/.conan/.conan/profiles/default
```
Again it creates nested ".conan/" directory?
With that dir structure but setting CONAN_USER_HOME a bit differently:

```
export CONAN_USER_HOME=/home/foo/conan_home_test
conan install . -pr test_profile
ERROR: 1 Profile not found: /home/foo/conan_home_test/.conan/profiles/
```
Although that directory exists, but is seems to be missing the filename in the path?

What I'm doing wrong here to make it pick the given profile file from my custom CONAN_USER_HOME?

**Comment by uilianries** [comment]
Hi @unzap !

The Conan behavior is correct, [CONAN_USER_HOME](https://docs.conan.io/en/latest/reference/env_vars.html#conan-user-home) will define the base folder for conan cache folder actually. I think this is not clear enough in the documentation. I'll update it.

When you set `CONAN_USER_HOME`, Conan will create a folder named `.conan`, which contains all stuff like settings file, config file, packages, profiles ...

The best way that you can use to create a profile is using the command line, so you see the correct path:

    conan profile new foo --detect

You can obtain more details about profile command [here](https://docs.conan.io/en/latest/reference/commands/misc/profile.html).

Regards!

**Comment by unzap** [comment]
Hi @uilianries , thanks for the reply!

> Lists profiles in the ‘.conan/profiles’ folder, or shows profile details.

Based on your comment and documentation, is it not possible to define custom "CONAN_HOME" i.e. to have profiles (+settings etc.) in any other folder than $HOME/.conan?

Example use case being that there are two (or more) projects with differing settings.yml and profiles (profile filenames may conflict). When switching between those projects one would need to replace all files under $HOME/.conan?

**Comment by memsharded** [comment]
Yes, right now, the ``.conan`` subfolder is some legacy that we couldn't get rid of without breaking. This will be fixed in Conan 2.0, and CONAN_USER_HOME will be able to define a folder without the ``.conan`` subfolder.

If that would fix this issue, can we consider this a duplicate of https://github.com/conan-io/conan/issues/5295?

**Comment by unzap** [comment]
@memsharded that's good news, thanks!

> If that would fix this issue, can we consider this a duplicate of #5295?

yes :)
